src/Request.php

```diff
@@ -428,6 +511,28 @@ public function getRangeUnit(): ?string
         return null;
     }
 
+    /**
+     * Set GET parameters
+     *
+     * @param array $params
+     * @return void
+     */
+    public function setGet(array $params)
```

Comment:
Both `setGet` and `setPost` are coupled with legacy PHP interface and would be better to use the relevant HTTP terminology we're using across the library. I would rather name them as `setQueryString` and `setPayload` and overwrite existing internal attributes. We might need to add a $queryString property like we have one for payload and adapt the `generateInput()` method to use it properly.